packages/fxa-settings/src/lib/sensitive-data-client.ts

```diff
@@ -6,6 +6,20 @@
 
 export const AUTH_DATA_KEY = 'auth';
 
+export type SensitiveDataClientData = {
+  [AUTH_DATA_KEY]: {
+    emailForAuth?: string;
```

Comment:
Could something like provide better type safety?

  ```
  export type SensitiveDataClientData = {
    auth: 
      | { emailForAuth:string, authPW:string }
      | { keyFetchToken: string; unwrapBKey: string; }
  };
  ```
  
This is based on hindsight, where we had the issue that an unwrapBKey was missing, and this ended up being hard to track down. Sometimes we play might be playing a little fast and loose with optional fields.

Reply:
> the main question is there a safer way to model this

> I'm also noticing that there's not a lot of type safety when calling setData. Any key value can be passed and typescript won't complain.

👍 I was planning on filing a follow up issue exactly because of this reason. I'll see if I can modify it a little bit though.